classes/app.rb

```diff
@@ -1,119 +1,287 @@
+  end
+
+  def find_or_create_classroom(label)
+    classroom = @classrooms.find { |c| c.label == label }
+    unless classroom
+      classroom = Classroom.new(label)
+      @classrooms << classroom
+    end
+
+    classroom
+  end
+
+  def create_book
+    puts 'Title:'
+    title = gets.chomp
+    puts 'Author:'
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books << book
+    puts 'Book created successfully.'
+  end
+
+  def create_rental
+    puts 'Select a person from the following list by number (not id):'
+    list_people_with_numbers
+    person_number = gets.chomp.to_i
+
+    selected_person = @people[person_number - 1]
+
+    if selected_person
+      puts 'Select a book from the following list by number'
+      list_books_with_numbers
+      book_number = gets.chomp.to_i
+
+      selected_book = @books[book_number - 1]
+
+      if selected_book
+        puts 'Date (YYYY-MM-DD):'
+        gets.chomp
+
+        begin
+          date = Time.now
+          rental = Rental.new(date, selected_book, selected_person)
+          @rentals << rental
+          puts 'Rental created successfully'
+        rescue ArgumentError
+          puts 'Invalid date format. Please use YYYY-MM-DD.'
+        end
       else
-        puts "Invalid choice. Please enter a valid option. (#{parent_permission})"
-        return
+        puts 'Invalid book selection'
       end
-      @newpersion = Student.new('Unknown', age, name, parent_permission: true)
     else
-      specialization = UserInput.new.user_input_as_string('Please enter a Specialization:')
-      @newpersion = Teacher.new(specialization, age, name)
+      puts 'Invalid person selection'
+    end
+  end
 
+  def list_books_with_numbers
+    @books.each_with_index do |book, index|
+      puts "#{index + 1}. #{book.title} by #{book.author}"
     end
-    @people.push(@newpersion)
-    puts 'Person created successfully'
   end
 
-  def create_book
-    print 'Title: '
-    title = gets.chomp.to_s
+  def list_people_with_numbers
+    @people.each_with_index do |person, index|
+      puts "#{index + 1}. #{person.name} (ID: #{person.id})"
+    end
+  end
+
+  def list_rentals_for_person
+    puts 'ID of person:'
+    person_id = gets.chomp.to_i
+
+    person = @people.find { |p| p.id == person_id }
+
+    if person
+      puts 'Rentals:'
+      person.rentals.each do |rental|
+        puts " Date: #{rental.date.strftime('%y-%m-%d')}, Book: #{rental.book.title} by #{rental.book.author} "
+      end
+    else
+      puts 'Person not found.'
+    end
+  end
 
-    print 'Author: '
-    author = gets.chomp.to_s
+  def save_data
+    save_books
+    save_people
+    save_rentals
+  end
 
-    @books.push(Book.new(title, author))
-    puts 'Book created successfully'
+  def load_data
+    load_books
+    load_people
+    load_rentals
+  rescue Errno::ENOENT
+    # Handle the case when one or more files are missing
+    puts 'One or more data files are missing. Starting with empty data.'
```

Comment:
- [ ] As stated in the project requirement Good Job for Implementing  On exit from your app save all the data (books, people, and rentals) into JSON files as shown for sample I created for peoples and books shown with the below screenshot but After I quit the app and then  On startup of your app should load all data from files, But In the case of your app doesn't load the data I created for the sample peoples and books So Kindly Fix the issues.

-  On exit from your app it save data 
![image](https://github.com/ethred/OOP_schoo_library/assets/93598189/12f68bf0-225a-4a3c-84aa-e4b40d3371b6)

![image](https://github.com/ethred/OOP_schoo_library/assets/93598189/8bb7767b-dc21-4261-a582-5f356358ae8e)

- But I after exit the app and startup of your app , it doesn't load data . Your app reset to default as shown with below screenshot
![image](https://github.com/ethred/OOP_schoo_library/assets/93598189/baf2efae-2957-4d6f-8153-e835e36adf3f)